diem-node/src/lib.rs

```diff
@@ -415,6 +435,32 @@ pub fn setup_environment(node_config: &NodeConfig, logger: Option<Arc<Logger>>)
     // TODO set up on-chain discovery network based on UpstreamConfig.fallback_network
     // and pass network handles to mempool/state sync
 
+    // For now, spawn all of the storage-service servers on the same runtime.
+    let storage_service_rt = Builder::new_multi_thread()
+        .thread_name("storage-service-servers")
+        .enable_all()
+        .build()
+        .expect("Failed to start the DiemNet storage-service runtime.");
+    let storage_reader = StorageReader::new(Arc::clone(&db_rw.reader));
+    for storage_service_events in storage_service_server_network_handles {
+        let storage_service_server = StorageServiceServer::new(
+            storage_service_rt.handle().clone(),
+            storage_reader.clone(),
+            storage_service_events,
+        );
+        storage_service_rt.spawn(storage_service_server.start());
+    }
+
+    // Combine all storage service client handles
+    let storage_service_client = StorageServiceClient::new(
+        StorageServiceMultiSender::new(storage_service_client_network_handles),
+        peer_metadata_storage.clone(),
+    );
+    let (_diemnet_data_client, _data_summary_poller) =
+        DiemNetDataClient::new(TimeService::real(), storage_service_client);
+    // TODO(philiphayes): Wire up data client with state-sync-v2 and spawn poller.
+    // state_sync_v2_rt.spawn(_data_summary_poller);
```

Comment:
I still think it's odd that we need clients of the `DiemNetDataClient` to spawn the poller task. Why not make the `DiemNetDataClient` responsible for spawning this task on creation? e.g., give it a runtime reference (like the storage service runtime) and spawn the task on the runtime?